src/plugins/data/public/ui/dataset_selector/advanced_selector.tsx

```diff
@@ -48,22 +56,38 @@ export const AdvancedSelector = ({
         }),
     },
   ]);
-  const [selectedDataset, setSelectedDataset] = useState<BaseDataset | undefined>();
 
-  return selectedDataset ? (
+  const [currentSelectedDataset, setCurrentSelectedDataset] = useState<BaseDataset | undefined>(
+    selectedDataset
+  );
+
+  return currentSelectedDataset ? (
     <Configurator
-      baseDataset={selectedDataset}
+      baseDataset={currentSelectedDataset}
```

Comment:
what is the difference between selectedDataset and currentSelectedDataset? Are there more accurate names?

Reply:
+1. I think this might be causing some incorrect states on my local